crates/ol/mempool/src/state.rs

```diff
@@ -0,0 +1,1933 @@
+//! Mempool service state management.
+
+use std::{
+    collections::{BTreeMap, BTreeSet, HashMap, HashSet},
+    sync::Arc,
+};
+
+use ssz::Decode;
+use strata_acct_types::{AccountId, AccountTypeId};
+use strata_db_types::types::MempoolTxData;
+use strata_identifiers::{OLBlockCommitment, OLTxId};
+use strata_ledger_types::{IAccountState, ISnarkAccountState, IStateAccessor};
+use strata_ol_chain_types_new::{OLBlock, OLTransaction, TransactionPayload};
+use strata_ol_state_types::OLState;
+use strata_service::ServiceState;
+use strata_storage::NodeStorage;
+
+use crate::{
+    OLMempoolError, OLMempoolResult,
+    ordering::{FifoOrderingStrategy, OrderingStrategy},
+    types::{
+        MempoolEntry, MempoolOrderingKey, OLMempoolConfig, OLMempoolRejectReason, OLMempoolStats,
+        OLMempoolTransaction, OLMempoolTxPayload,
+    },
+    validation::{BasicTransactionValidator, TransactionValidator},
+};
+
+/// Immutable context for mempool service (shared via Arc).
+pub(crate) struct MempoolContext {
+    /// Mempool configuration.
+    pub(crate) config: OLMempoolConfig,
+
+    /// Storage backend for creating StateAccessor instances.
+    /// Mempool creates StateAccessor per-operation based on current_tip.
+    pub(crate) storage: Arc<NodeStorage>,
+
+    /// Ordering strategy for transaction priority.
+    pub(crate) ordering_strategy: Arc<dyn OrderingStrategy>,
+
+    /// Transaction validator for validation strategies.
+    /// Uses concrete type since TransactionValidator trait is not object-safe (has generic
+    /// method).
+    pub(crate) validator: Arc<dyn TransactionValidator>,
+}
+
+impl std::fmt::Debug for MempoolContext {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("MempoolContext")
+            .field("config", &self.config)
+            .field("ordering_strategy", &"<OrderingStrategy>")
+            .field("storage", &"<NodeStorage>")
```

Comment:
Including these fields in the debug string doesn't provide any information, can just remove them.